File: Python/Mahasiswa.py
# Mahasiswa adalah bagian dari sivitas akademik

from SivitasAkademik import SivitasAkademik

class Mahasiswa(SivitasAkademik):
    def __init__(self, nim="-", prodi="-", fakultas="-"):
        super().__init__()      # memanggil constructor parent nya
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas

    def setNim(self, nim):
        self.__nim = nim

    def getNim(self):
        return self.__nim
	
    def setProdi(self, prodi):
        self.__prodi = prodi
	
    def getProdi(self):
        return self.__prodi
	
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas
	
    def getFakultas(self):
        return self.__fakultas
	

File: cpp/Mahasiswa.cpp
//Mahasiswa adalah bagian dari sivitas akademik

class Mahasiswa : public SivitasAkademik
{
	private:
		string nim;
		string prodi;
		string fakultas;
	public:
		Mahasiswa(){
			this->nim = "-";
			this->prodi = "-";
			this->fakultas = "-";
		}
		void setNim(string nim){
			this->nim = nim;
		}
		string getNim(){
			return this->nim;
		}
		void setProdi(string prodi){
			this->prodi = prodi;
		}
		string getProdi(){
			return this->prodi;
		}
		void setFakultas(string fakultas){
			this->fakultas = fakultas;
		}
		string getFakultas(){
			return this->fakultas;
		}
};